src/core/symbology/qgssymbollayerutils.cpp

```diff
@@ -5096,3 +5097,234 @@ QgsStringMap QgsSymbolLayerUtils::evaluatePropertiesMap( const QMap<QString, Qgs
   }
   return properties;
 }
+
+QSize QgsSymbolLayerUtils::tileSize( int width, int height, double &angleRad )
+{
+
+  // Precondition
+  Q_ASSERT( angleRad >= 0 && angleRad < M_PI * 2 );
```

Comment:
This is not so nice for pyqgis use -- couldn't we just mod the value automatically to ensure this condition is met?

Reply:
Ok, I'll move the logic from client code into this method